developer/js/lexical-model-compiler/build-trie.ts

```diff
@@ -65,6 +75,58 @@ export function parseWordList(contents: string): WordList {
   return result;
 }
 
+export function compileWordListCharacterSet(wordlist: WordList, searchTermToKey?: (wf: string) => string): string {
+  let charMap: {[char: string]: number} = {};
+  let charList: string[] = [];
+
+  for(let [word, weight] of wordlist) {
+    // There's a problem if we have `null` or `undefined` for a word.
+    if(!word) {
+      throw TypeError();
+    }
+
+    // We want to operate on the search-term form, since that's what is used to look up the words in the Trie.
+    if(searchTermToKey) {
+      word = searchTermToKey(word);
+    }
+
+    for(let i=0; i < word.length; i++) {
+      let code = word.charCodeAt(i);
+      let char = word.charAt(i);
+
+      // Check for surrogate pairs
+      if(code >= 0xD800 && code <= 0xDBFF && word.length > i+1) {
+        let pairedCode = word.charCodeAt(i+1);
+
+        // Ensure it's actually a legit paired code
+        if(pairedCode >= 0xDC00 && pairedCode <= 0xDFFF) {
+          // Computes the actual character's code.
+          //code = (code - 0xD800) * 0x400 + pairedCode - 0xDC00 + 0x10000
+
+          // Only process the second code if it's actually paired.
+          char = char + word.charAt(++i); // The second code is processed, so skip it in the loop.
```

Comment:
Put the ++i on its own line.

```suggestion
          ++i;  // The second code is processed, so skip it in the loop.
          char = char + word.charAt(i);
```

Reply:
Is there some code convention thing I'm missing with this?  I mean, I can do it, but I don't quite get the aversion.

Reply:
It's in an effort to have one operation per line, i.e., incrementing the index and calling the `.charAt()` of the string object. In C, this practice is used a lot, but when I'm reading a statement like this one, I have multiple things to process on one line. Ideally, I'd like only one thing to process per line. [I'm not the only one that has this sentiment](https://stackoverflow.com/a/971442/6626414).

JSLint claims that the `++` and `--` operators [should be banned outright](https://www.jslint.com/help.html#inc) because of their potential to be used to "creatively"; I think that's overdoing it though.